src/build.ts

```diff
@@ -74,6 +75,23 @@ export async function build(
   let assetCount = 0;
   let pageCount = 0;
   const pagePaths = new Set<string>();
+
+  const {title} = config;
+  const buildManifest: BuildManifest = {...(title && {title}), pages: [], modules: [], files: []};
+  const addToManifest = (
+    type: string,
+    file: string,
+    {title, path, params}: {title?: string | null; path?: string; params?: Params}
+  ) => {
+    const source = path == null || path === file.slice(1) ? null : join("/", path);
+    buildManifest[type].push({
+      path: config.normalizePath(file),
+      ...(title != null && {title}),
+      ...(source && {source}),
+      ...(params && {params})
```

Comment:
we don't strictly need to write the params since they can be derived from the file + source by pattern matching

should I remove them?

Reply:
It'd be nice to include them here, since you have them. It saves us work later, and I'm not worried about the extra file size.

Reply:
My concern here is that we expose a bit more API surface, so we now have to wonder _e.g._ what happens if someone uses a modified version of framework that creates a discrepancy. (And in any case there is work to do to validate the inputs.)

Reply:
That's a good point. Either way, we don't need this now so we could go with the more minimal version at first.